Documentation for new Raspberry PI camera platform "rpicam" (#8177)
home-assistant/home-assistant.github.io: source/_components/camera.rpicam.markdown

---
layout: page
title: "MQTT Raspberry PI camera"
description: "Instructions on how to integrate Raspberry Pi cameras within Home Assistant."
date: 2019-01-15 8:15
sidebar: true
comments: true
sharing: true
footer: true
ha_category: Camera
logo: home-assistant.png
ha_release: 0.85
ha_iot_class: "depends"
---

> The `rpicam` camera platform allows to control a raspberry pi camera through mqtt commands.

Reviewer: ```suggestion
The `rpicam` camera platform allows controlling a raspberry pi camera through MQTT commands.
```

> This is a third-party integration for homeassistant that provides further control from HA for a raspberry PI camera based on RPi-Cam-Web-Interface (`https://elinux.org/RPi-Cam-Web-Interface`).

Reviewer: ```suggestion
This is a third-party integration for Home Assistant that provides further control from HA for a raspberry PI camera based on RPi-Cam-Web-Interface (`https://elinux.org/RPi-Cam-Web-Interface`).
```

## {% linkable_title Configuration %}

It can be configured like a `generic` IP camera and supports the same configuration parameters:

    https://www.home-assistant.io/components/camera.generic/

> Set `rpicam` (instead of `generic`) as platform and follow the link above for further configuration details.

Reviewer: ```suggestion
Set `rpicam` (instead of `generic`) as a platform and follow the link above for further configuration details.
```

Configuration example:

```yaml
      - platform: rpicam
        name: backyard
        still_image_url: http://192.168.1.12/cam.jpg

>         username: <USER>

Reviewer: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Author: This new platform reuses the same configuration details as the `generic IP camera` platform and I'm going to fix the documentation to reflect the document it comes from. Unfortunately the camera name, is not optional in this case and I need to leave it in the example for consistency with the platform configuration.
I'm going to review the code as well, to ensure the constraint is enforced.

>         password: !secret <USER_PASS>

Reviewer: Same as above

```

## {% linkable_title What provides %}

This platform uses MQTT for interaction with the remote raspberry PI where an agent is configured and running (`https://gitlab.com/gbus/rpi-cam-mqtt`).
You can:

    - enable/disable the camera
    - take pictures/videos
    - enable/disable motion detection
    - change exposure mode
    - pan/tilt the camera

> ## {% linkable_title What it dosn't provide %}

Reviewer: ```suggestion
## {% linkable_title What it doesn't provide %}
```

> ha-rpicam doesn't send or receive streaming data through MQTT. This is a specific solution to only control the behaviour of the camera as an automation component under HA. Commands to capture images/videos are equivalent to pushing the camera web interface buttons. The captured media content would still be accessible as normal from the download area of the web interface itself.

Reviewer: ```suggestion
`ha-rpicam` doesn't send or receive streaming data through MQTT. This is a specific solution to only control the behavior of the camera as an automation component under HA. Commands to capture images/videos are equivalent to pushing the camera web interface buttons. The captured media content would still be accessible as normal from the download area of the web interface itself.
```
